### PR TITLE
example of how to do a statistical performance regression test

### PR DESCRIPTION
**Pull request by bengland2**
plan to make this callable, but
it can be run today using input files containing samples

### CODE REVIEW

File: src/touchstone/utils/t_test.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+# numpy and scipy RPMs
+# for python3, there are python3-{numpy,scipy} RPMs
+#
+
+import os
+import sys
+from sys import argv, exit
+import math
+import numpy
+import scipy
```
Comment (aakarshg) [inline src/touchstone/utils/t_test.py:46]: we'll need to add these to the https://github.com/cloud-bulldozer/touchstone/blob/master/requirements.txt

**Comment by jtaleric** [comment]
@bengland2 doing so late spring cleaning, do we want to keep this around?

**Comment by bengland2** [comment]
@jtaleric This dates back to a disagreement with aakarsh a year ago.   Does benchmark-comparison (formerly touchstone)  do a statistically valid comparison of 2 sample sets?   It's not valid to conclude that two sample sets are different by just comparing sample set means, you have to incorporate variance of sample sets somehow.   I'd have to look closer at benchmark-comparison to see what it does.

**Comment by mfleader** [comment]
@bengland2 Has a point about comparing two sets of samples on an index value like a mean. However, use of the frequentist t-test on any given pair of samples of continuous data, is not necessarily justified here.

**Comment by jtaleric** [comment]
> @jtaleric This dates back to a disagreement with aakarsh a year ago. Does benchmark-comparison (formerly touchstone) do a statistically valid comparison of 2 sample sets?

Today, it doesn't do inter-uuid comparisons to determine variance between samples.

We would need to define the sample field to do the variance comparison on. We could make some default assumptions, for tings like FIO, but we would want to allow some generalizations too. 

Instead of a stand-alone script, I think it might make more sense to have this part of the `gen_result_dict` method? 


> It's not valid to conclude that two sample sets are different by just comparing sample set means, you have to incorporate variance of sample sets somehow. I'd have to look closer at benchmark-comparison to see what it does.

Understood. We just need defined ways to capture and compare the samples. 

I think we can close this for now, and start a RFE / Issue to discuss how to implement something like this? Thoughts?

**Comment by mfleader** [comment]
I'm super interested in working on this as soon I get the cycles. I feel like @whitleykeith has also mentioned being interested in this and how it can integrate with the dashboard.

**Comment by jtaleric** [comment]
> I'm super interested in working on this as soon I get the cycles. I feel like @whitleykeith has also mentioned being interested in this and how it can integrate with the dashboard.

I am not sure I have explained myself well here... There is quite a bit of context you would need in order to tackle this. Not every workload would have "sample".

**Comment by mfleader** [comment]
That is a good point, workloads would need to output the raw data, and not summary statistics like the mean, median, p99, or standard deviation. @bengland2 and I have talked a little bit about this, and some workloads, like FIO, would need to output a histogram of a run's results to facilitate estimating dispersion and extreme values, like p99. Outside of that, we can talk about workloads that do not make raw data available, and strategize potential solutions.

**Comment by bengland2** [comment]
I retired.

**Pull request closed without merging**